lib/full-site-editing/edit-site-page.php

```diff
@@ -101,7 +101,7 @@ function gutenberg_edit_site_init( $hook ) {
 			'__experimentalBlockPatternCategories' => WP_Block_Pattern_Categories_Registry::get_instance()->get_all_registered(),
 		)
 	);
-	$settings = gutenberg_experimental_global_styles_settings( $settings );
+	$settings = apply_filters( 'block_editor_settings_all', $settings );
```

Comment:
I remove `gutenberg_experimental_global_styles_settings` because it is already being added as a callback to the `block_editor_settings_all` filter in `lib/global-styles.php`

https://github.com/WordPress/gutenberg/blob/1c733c07a4f0d48d5f594c0af44047ac11d5347a/lib/global-styles.php#L329-L334

Reply:
We'll need to add a second param to the `apply_filters` call here, as it's called with `WP_Block_Editor_Context` in the second param [here in Core](https://github.com/WordPress/WordPress/blob/79685db062fa5efa5c31f133f177a0d58eef22b9/wp-includes/block-editor.php#L362-L370). 

It appears that, if we're not in a post editing context, we're just using `new WP_Block_Editor_Context()` with no params, like [here in the Widgets Block Editor](https://github.com/WordPress/gutenberg/blob/339a7d783ad53dc9d437b3b5764714843caea2b8/lib/widgets-page.php#L81). 

So sum up, we should have:

```php
$site_editor_context = new WP_Block_Editor_Context();
$settings = apply_filters( 'block_editor_settings_all', $settings, $site_editor_context );
```

Oh! Also calls to `apply_filters` are always documented. Since `block_editor_settings_all` is already documented in Core, referencing that is sufficient. Now we get:


```php
$site_editor_context = new WP_Block_Editor_Context();
/** This filter is documented in wp-includes/wp-block-editor.php */ 
$settings = apply_filters( 'block_editor_settings_all', $settings, $site_editor_context );
```

Reply:
Awesome callouts. Thanks @mattwiebe